trulens_eval/trulens_eval/utils/imports.py

```diff
@@ -682,3 +684,104 @@ def __exit__(self, exc_type, exc_value, exc_tb):
             raise ImportError(self.messages.import_error) from exc_value
 
         # Exception will be propagated unless we return True so we don't return it.
+
+
+def get_modules(path: Path, startswith: Optional[str] = None) -> Iterable[str]:
+    """Get all module names in the given path that start with the given
+    prefix.
+    
+    Args:
+        path: The path to search for modules.
+
+        startswith: If given, only modules that with this prefix are returned.
+
+    Returns:
+        Iterable of module names. These are fully qualified.
+    """
+
+    for modinfo in pkgutil.iter_modules([str(path)]):
+
+        if startswith is not None and not modinfo.name.startswith(startswith):
+            continue
+
+        yield modinfo.name
+
+        if modinfo.ispkg:
+            for submod in get_modules(path / modinfo.name, startswith=None):
+                submodqualname = modinfo.name + "." + submod
+
+                if startswith is not None and not submodqualname.startswith(
+                        startswith):
+                    continue
```

Comment:
how could this ever happen? It seems that given `submodqualname = modinfo.name + "." + submod` and we know from L704 if `startswith is not None` then `modinfo.name.startswith(startswith)` is `True`.

Reply:
fixed